sklearn/utils/__init__.py

```diff
@@ -1182,3 +1204,67 @@ def is_abstract(c):
     # itemgetter is used to ensure the sort does not extend to the 2nd item of
     # the tuple
     return sorted(set(estimators), key=itemgetter(0))
+
+
+def _safe_tags(estimator, key=None, default=None):
+    """Safely get estimator tags for common checks.
+
+    :class:`~sklearn.BaseEstimator` provides the estimator tags machinery.
+    However, if a compatible estimator does not inherit from this base class,
+    we should default to the default tag.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator from which to get the tag.
+
+    key : str, default=None
+        Tag name to get. By default (`None`), all tags are returned.
+
+    default : list of {str, dtype} or bool, default=None
+        `default` allows to define the `default` value of a tag if it is not
+        present in `_DEFAULT_TAGS` or to overwrite the value in `_DEFAULT_TAGS`
+        if it the tag is defined. When `default is None`, no default values nor
+        overwriting will take place. If `default is not None` but that the
+        tag is defined, `default` will be discarded.
+
+    Returns
+    -------
+    tags : dict
+        The estimator tags.
+    """
+    if hasattr(estimator, "_get_tags"):
```

Comment:
This is a minor-ish nit. I usually think that writing code with less indents be simplify to follow.

```py
if hasattr(estimator, "_get_tags"):
	if key is None:
		tags = estimator._get_tags()
		return {
			key: tags.get(key, _DEFAULT_TAGS[key])
            for key in _DEFAULT_TAGS.keys()
        }

	if default is not None:
		return estimator._get_tags().get(key, default)

	try:
		return estimator._get_tags().get(key, _DEFAULT_TAGS[key])
	except KeyError as exec:
		...

# not hasattr(estimator, "_get_tags")
if key is None:
	return _DEFAULT_TAGS

if default is not None:
	return default

try:
	return _DEFAULT_TAGS[key]
except KeyError as exc:
	...
```

Reply:
I make it this way because @NicolasHug finds it more explicit. To be honest, I don't have preferences. We will need a third reviewer to choose either way :)

Reply:
Oh no. I would not block this PR because of this.

(I usually choose this style because of our 80 line limit and the default 4-space indentation.)